src/components/AppSettingsDialog/AppSettingsDialog.vue

```diff
@@ -0,0 +1,260 @@
+			const navigationItems = slots.filter(vNode => vNode.componentOptions).map(vNode => vNode.componentOptions.propsData.sectionTitle)
+			// Check for the uniqueness of section titles
+			navigationItems.forEach((element, index) => {
+				const newArray = [...navigationItems]
+				newArray.splice(index, 1)
+				if (newArray.indexOf(element) !== -1) {
+					throw new Error(`Duplicate section title found: ${element}. Settings navigation sections must have unique section titles.`)
+				}
+			})
+			return navigationItems
+		},
+
+		/**
+		 * Scrolls the content to the selected settings section.absolute
+		 * @param {string} item the name of the section
+		 */
+		handleSettingsNavigationClick(item) {
+			this.selectedSection = item
+		},
+
+		handleCloseModal() {
+			this.$emit('update:open', false)
+		},
+	},
+
+	render(createElement) {
+
+		/**
+		 * Build the navigation
+		 * @returns {object} the navigation
+		 */
+		const createAppSettingsNavigation = () => {
+			if (this.hasNavigation) {
+				return [createElement('div', {
+					attrs: {
+						class: 'app-settings__navigation',
+					},
+				}, [createElement('ul', {
+					attrs: {
+						class: 'navigation-list',
+					},
+				}, this.getSettingsNavigation(this.$slots.default).map(item => {
+					return createListElemtent(item)
+				}))])]
+			} else {
+				return []
+			}
+		}
+
+		/**
+		 * Build each list element in the navigation
+		 * @param {object} item the navigation item
+		 * @returns {object} the list element
+		 */
+		const createListElemtent = (item) => createElement('li', {}, [createElement('a', {
+			class: {
+				'navigation-list__link': true,
+				'navigation-list__link--active': item === this.selectedSection,
+			},
+
+			on: {
+				click: () => this.handleSettingsNavigationClick(item),
+			},
+		}, item)])
+
+		// Return value of the render function
+		if (this.open) {
+			return createElement('Modal', {
+				on: {
+					close: () => { this.handleCloseModal() },
+				},
+			}, [
+				createElement('div', {
+					attrs: {
+						class: 'app-settings',
+					},
+				}, [...createAppSettingsNavigation(),
+					createElement('div', {
+						attrs: {
+							class: 'app-settings__content',
+						},
+					}, this.$slots.default.filter(element => {
+						if (this.isMobile) {
+						// Show all the sections if mobile
+							return true
+						}
+						if (!element.componentOptions) {
+							return false
+						}
+						return element.componentOptions.propsData.sectionTitle === this.selectedSection
+					}
+					))]),
+			])
+		} else {
+			return undefined
+		}
+	},
+}
+
+</script>
+
+<style lang="scss" scoped>
+.app-settings {
+	padding: 28px 16px;
+	display: flex;
+	height: 600px;
+	width: 100%;
+	&__navigation {
+		width: 200px;
+		margin-right: 20px;
+	}
+	&__content {
+		width: 400px;
+		overflow-y: auto;
+		overflow-x: hidden;
+		word-wrap: break-word;
+		padding: 0 20px;
+	}
+}
+
+.navigation-list {
+	&__link {
+		display: block;
+		font-size: 16px;
+		height: 44px;
```

Comment:
Please use the available variables

```suggestion
		height: $clickable-area;
```